### PR TITLE
#738: Send email verification link

### PR DESCRIPTION
**Pull request by bdragan**
#738: Send email verification link.

### CODE REVIEW

**Comment by karato** [comment]
@bdragan Thanks for the pull request, let me find a reviewer..

**Comment by karato** [comment]
@caarlos0 review it

File: netbout-web/src/main/java/com/netbout/dynamo/DyAlias.java [hunk 1]
```diff
@@ -113,7 +113,9 @@ public String email() throws IOException {
 
     @Override
     public void email(final String email) throws IOException {
-        if (!email.matches("[a-z_\\.@\\-A-Z0-9]+")) {
+        final String pattern = "[a-z_\\.@\\-A-Z0-9]";
```
Comment (caarlos0) [inline netbout-web/src/main/java/com/netbout/dynamo/DyAlias.java]: @bdragan performance-wise I think it's better to compile the regex and keep it in a constant...

File: netbout-web/src/main/java/com/netbout/email/EmAlias.java [hunk 2]
```diff
@@ -119,6 +125,48 @@ public void email(final String email, final String urn, final Bout bout)
...
+     *  change your email address", etc.
+     */
+    @Override
+    public void email(final String email, final String link)
+        throws IOException {
+        this.email(email);
+        try {
+            this.postman.send(
+                new Envelope.MIME()
+                    // @checkstyle IndentationCheck (12 lines)
```
Comment (caarlos0) [inline netbout-web/src/main/java/com/netbout/email/EmAlias.java]: @bdragan why?

**Comment by caarlos0** [comment]
@bdragan couple of comments

File: netbout-web/src/main/java/com/netbout/email/EmAlias.java (hunk 2 above)
Comment (bdragan) [inline netbout-web/src/main/java/com/netbout/email/EmAlias.java]: @caarlos0 I found it difficult to format correctly, maybe it has something to do with [this] checkstyle issue. However, I moved subject `with` clause one level up (before the recipient); it seems that checkstyle is happy now. :)

**Comment by bdragan** [comment]
@caarlos0 Thanks, please take a look at new changes.

File: netbout-web/src/main/java/com/netbout/email/EmAlias.java (hunk 2 above)
Comment (caarlos0) [inline netbout-web/src/main/java/com/netbout/email/EmAlias.java]: @bdragan nice, thanks

**Comment by caarlos0** [comment]
@bdragan thanks

**Comment by caarlos0** [comment]
@rultor merge

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by karato** [comment]
@ypshenychka please, review this issue for QA compliance, as per [par.24](http://at.teamed.io/policy.html#24)

**Comment by karato** [comment]
@rultor deploy now pls

**Comment by ypshenychka** [comment]
@bdragan 
According to our [QA Rules](http://www.teamed.io/qa.html):
>Pull request description explains the solution proposed and contains a link to the original ticket it is related to.

Pull request description must briefly explain what was done in the branch, so that we can use it for future reference. Please extend description by providing more information.

**Comment by ypshenychka** [comment]
@caarlos0 
According to our [QA Rules](http://www.teamed.io/qa.html):
>The code reviewer found at least three problems in the code.

Two issues were found during code review. 
Please confirm that you'll try to find at least three problems while future reviews.

**Comment by caarlos0** [comment]
@ypshenychka I confirm

**Comment by ypshenychka** [comment]
@caarlos0 thanks

**Comment by ypshenychka** [comment]
@bdragan ping

**Comment by bdragan** [comment]
@ypshenychka reply :)

**Comment by caarlos0** [comment]
@bdragan you should improve your PR description...

**Comment by bdragan** [comment]
@ypshenychka @caarlos0 Ups, sorry, thanks for reminding me. PR description improved.

**Comment by ypshenychka** [comment]
@bdragan Thank you.

**Comment by ypshenychka** [comment]
@karato Quality is acceptable.

**Comment by karato** [comment]
> @karato Quality is acceptable.

@ypshenychka thanks for the review, we'll try better next time

**Comment by karato** [comment]
@caarlos0 I added 10 mins to @ypshenychka (for QA review) in transaction 70639080. Much appreciated! **23 mins** was added to your account, payment ID `565600c25015d269dd000114`, time spent is 33 hours and 42 mins. you're getting extra minutes here (c=8). added +23 to your rating, now it is equal to [+744](http://www.netbout.com/b/35095?open=rating)

**Pull request merged**